frontend/src/components/input.tsx

```diff
@@ -2,59 +2,78 @@ import React, {
   ChangeEvent,
   FormEvent,
   useCallback,
-  useMemo,
   useState,
+  useLayoutEffect,
+  useRef,
 } from 'react';
 import styles from './input.module.css';
-import RightArrow from '../icons/map-arrow-right.svg';
-import classNames from 'classnames';
+import RightArrowIcon from '../icons/send.svg';
+import UploadIcon from '../icons/upload.svg';
 import { Suggestions } from './suggestions';
+import { Button } from './button';
 
 export interface InputProps {
   sendMessage: (message: string) => void;
-  suggestions: string[];
+  waiting: boolean;
 }
 
-export const Input = ({ sendMessage, suggestions }: InputProps) => {
+export const Input = ({ sendMessage, waiting }: InputProps) => {
   const [userInput, setUserInput] = useState<string>('');
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
 
-  const onChange = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+  const onChange = useCallback((event: ChangeEvent<HTMLTextAreaElement>) => {
     setUserInput(event.target.value);
   }, []);
 
+  useLayoutEffect(() => {
+    if (textareaRef.current) {
+      textareaRef.current.style.height = 'auto';
+      const textareaHeight = textareaRef.current.scrollHeight;
+
+      if (textareaHeight > 110) {
+        textareaRef.current.style.height = '110px';
+        textareaRef.current.style.overflowY = 'auto';
+      } else {
+        textareaRef.current.style.height = `${textareaHeight}px`;
+        textareaRef.current.style.overflowY = 'hidden';
+      }
+    }
+  }, [userInput]);
+
   const onSend = useCallback(
     (event: FormEvent<HTMLElement>) => {
       event.preventDefault();
-      sendMessage(userInput);
-      setUserInput('');
+      if (!waiting && userInput.trim().length > 0) {
+        sendMessage(userInput);
+        setUserInput('');
+      }
     },
-    [sendMessage, userInput],
+    [sendMessage, userInput, waiting],
   );
 
-  const sendDisabled = useMemo(() => userInput.length === 0, [userInput]);
-
   return (
     <>
       <form onSubmit={onSend} className={styles.inputContainer}>
-        <input
-          className={styles.input}
-          onChange={onChange}
-          onSubmit={onSend}
-          placeholder="Send a Message..."
-          type="text"
-          value={userInput}
-        />
-        <button
-          className={classNames(styles.sendButton, {
-            [styles.disabled]: sendDisabled,
-          })}
-          onClick={onSend}
-          disabled={sendDisabled}
-        >
-          <img src={RightArrow} />
-        </button>
+        <div className={styles.parentDiv}>
+          <textarea
+            className={styles.textarea}
+            ref={textareaRef}
+            placeholder="Send a Message..."
+            value={userInput}
+            onChange={onChange}
+            rows={1}
+          />
+          <div className={styles.uploadButton_container}>
+            <button className={styles.uploadButton} disabled={true}>
```

Comment:
don't bother changing it here as we'll soon change this but you can just do `disabled` instead of `disabled={true}`